Drone/drone_base.py

```diff
@@ -0,0 +1,361 @@
+                    break
+
+                current_altitude = self.altitude()
+
+                if current_altitude >= target_altitude*0.95: # Trigger just below target alt.
+                    break
+                elif current_altitude >= target_altitude*0.6:
+                    thrust = c.SMOOTH_TAKEOFF_THRUST
+
+                dkw.set_attitude(self.vehicle, thrust=thrust)
+                time.sleep(1)
+            else:
+                self.logger.error("Could not take off!")
+            
+            self.logger.info(threading.current_thread().name + ": Finished takeoff")
+            finished.set()
+
+        finished = threading.Event()
+        threading.Thread(target=takeoff_thread, name="TakeoffThread-" + str(DroneBase.cnt), 
+            args=(target_altitude, finished, stop_event)).start()
+        DroneBase.cnt += 1
+        return finished
+
+    def land(self):
+        """
+        Lands the drone on the ground
+
+        Parameters
+        ----------
+        None
+
+        Precondition:
+        ----------
+        The drone is connected and armed.
+
+        Postcondition:
+        ----------
+        The drone is on the groun (and ideally not smashed into 1000 pieces)
+        
+        Returns:
+        ----------
+        None
+        """
+        def land_thread(finished):
+            self.logger.info(threading.current_thread().name + ": Starting land")
+            while not self.vehicle.mode == VehicleMode(c.LAND_MODE):
+                self.vehicle.mode = VehicleMode(c.LAND_MODE)
+            while self.vehicle.armed:
+                pass
+
+            self.logger.info(threading.current_thread().name + ": Finished  land")
+            finished.set()
+
+        finished = threading.Event()
+        threading.Thread(target=land_thread, name="LandThread-" + str(DroneBase.cnt), 
+            args=(finished,)).start()
+        DroneBase.cnt += 1
+        return finished
+
+    def move(self, direction, distance, stop_event, velocity=c.DEFAULT_VELOCITY):
+        """
+        Moves the drone along a path.
+
+        Parameters
+        ----------
+        direction: UP, DOWN, LEFT, RIGHT, FORWARD, BACK (as defined in constants.py) 
+            The direction the drone should travel in
+        distance: Double
+            The distance in meters the drone should travel in the given direction
+        emergency_land: threading.Event()
+            Set when this movement needs to stop
+
+        Precondition:
+        ----------
+        The drone is already in the air and not moving anywhere else
+
+        Postcondition:
+        ----------
+        The drone has moved in the specified direction by the specified number of meters
+
+        Returns:
+        ----------
+        threading.Event
+        """
+
+        # Calculate duration to send velocity command based on distance and velocity
+        duration = int(distance / velocity)
```

Comment:
Why is duration an int here? If I wanted to go .99 meters at 1 m/s, then I would get a duration of 0!

Reply:
The reason is that the mavlink message is sent every second (see the note at http://python.dronekit.io/examples/guided-set-speed-yaw-demo.html#send-global-velocity). The value is being used like this:
`for _ in range(0, duration)`
around drone_base.py line 309 so a float would cause error.

I don't know if sending every second is a requirement or not - we can do some experimenting.